ops/testing.py

```diff
@@ -1518,14 +1518,63 @@ def add_layer(
                 if service.override not in ('merge', 'replace'):
                     raise RuntimeError('500 Internal Server Error: layer "{}" has invalid '
                                        '"override" value on service "{}"'.format(label, name))
-                if service.override != 'replace':
+                if service.override == 'replace':
+                    layer.services[name] = service
+                elif service.override == 'merge':
+                    if combine:
+                        self._merge_service(label, name, service)
+                    else:
+                        layer.service[name] = service
+                else:
                     raise RuntimeError(
                         'override: "{}" unsupported for layer "{}" service "{}"'.format(
                             service.override, label, name))
-                layer.services[name] = service
+
         else:
             self._layers[label] = layer_obj
 
+    def _merge_service(self, label, name: str, service: pebble.Service):
```

Comment:
I'd like to see this method live on the `pebble.Service` class (perhaps public) so we can e.g. just `service.merge(other)` from the add_layer method.  This also has the added benefit of keeping all the merge code together with the attributes that it operates on - making it easier to keep things in sync as service attributes evolve in the future.  This should make it a bit nicer to unit test as well.

Reply:
Or maybe keep it private - e.g. `def _merge`

Reply:
I could be convinced either way.

Reply:
done